src/Illuminate/Database/Schema/Grammars/ChangeColumn.php

```diff
@@ -119,17 +128,35 @@ protected static function getDoctrineColumn(Table $table, Fluent $fluent)
      */
     protected static function getDoctrineColumnChangeOptions(Fluent $fluent)
     {
-        $options = ['type' => static::getDoctrineColumnType($fluent['type'])];
+        $doctrineType = static::getDoctrineColumnType($fluent['type']);
 
-        if (in_array($fluent['type'], ['text', 'mediumText', 'longText'])) {
+        $options = ['type' => $doctrineType];
+
+        if ($doctrineType->getName() === Types::TEXT) {
             $options['length'] = static::calculateDoctrineTextLength($fluent['type']);
         }
 
+        if ($fluent['type'] === 'binary') {
+            $options['length'] = AbstractMySQLPlatform::LENGTH_LIMIT_BLOB;
+        }
+
         if ($fluent['type'] === 'char') {
             $options['fixed'] = true;
         }
 
-        if (static::doesntNeedCharacterOptions($fluent['type'])) {
+        if ($fluent['type'] === 'timestamp') {
+            $options['platformOptions'] = [
+                'version' => true,
+            ];
+        }
+
+        if ($fluent['type'] === 'jsonb') {
+            $options['platformOptions'] = [
+                'jsonb' => true,
```

Comment:
What does this do? Why is it needed?

Reply:
Only PostgreSQL supports `jsonb`:
* Maps `jsonb` to `json`: https://github.com/doctrine/dbal/blob/36b9ffc15c31598baa2a37b1abdd5b860d5c164f/src/Platforms/PostgreSQLPlatform.php#L1169
* Uses `jsonb` option: https://github.com/doctrine/dbal/blob/36b9ffc15c31598baa2a37b1abdd5b860d5c164f/src/Platforms/PostgreSQLPlatform.php#L1315-L1322